echo_server/lib/echo_server.ex

```diff
@@ -0,0 +1,42 @@
+defmodule EchoServer do
+  use GenServer
+
+  @moduledoc """
+  Documentation for `EchoServer`.
+  """
+
+  @doc """
+  Ping-pong.
+
+  ## Examples
+      iex> {:ok, pid} = EchoServer.start_link()
+      iex> EchoServer.echo(pid, :ping)
+      {:pong, :nonode@nohost}
+
+  """
+
+  # Client
+
+  def start_link(default \\ %{}) do
+    GenServer.start_link(__MODULE__, default)
+  end
+
+  def echo(pid, message) do
+    case message do
+      :ping -> GenServer.call(pid, message)
+      _ -> raise "Only :ping message is allowed!"
+    end
+  end
+
+  # Server (callbacks)
+
+  @impl true
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call(:ping, _from, state) do
+    {:reply, {:pong, node()}, state}
```

Comment:
Можно еще message назад выслать.
```suggestion
  def handle_call(:ping, {pid, _ref}, state) do
    send(pid, {:pong, node()})
    {:reply, {:pong, node()}, state}
```

Тестируется это при помощи [`assert_receive/3`](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#assert_receive/3).

Reply:
@am-kantox  Добавил `assert_receive`, [обновил тесты](https://github.com/dmitry-sidorov/elixir_course_2024_thinknetica/pull/2/commits/0b137c1a101e00d37abaa1f7e46bc022708b8a67)